core/src/sumeragi/main_loop.rs

```diff
@@ -231,12 +231,11 @@ fn gossip_transactions<F>(
     F: FaultInjection,
 {
     if last_sent_transaction_gossip_time.elapsed() > sumeragi.gossip_period {
-        let txs = state
-            .transaction_cache
-            .iter()
-            .take(sumeragi.gossip_batch_size as usize)
-            .cloned()
-            .collect::<Vec<_>>();
+        // Transactions are intentionally taken from the queue instead of the cache
```

Comment:
This could impact performance, did you run any comparisons?

Reply:
Not yet, i have problems running `tps` benchmark on `dev` branch.

Reply:
Result of `tps-dev` benchmark doesn't reveal any changes in the performance.

```
Before (fbc7d346):
time:   [381.34 tps 455.59 tps 529.53 tps]

After (81448627):
time:   [433.57 tps 503.22 tps 568.07 tps]
change: [+26.541% +9.4650% -11.143%] (p = 0.39 > 0.05)
No change in performance detected.
```